### PR TITLE
feat: support link text

### PR DESCRIPTION
**Pull request by xrutayisire**
Resolves [DT-2272](https://linear.app/prismic/issue/DT-2272/aadev-i-can-use-prismic-sdks-with-support-link-text)

## Types of changes

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

- Support link text

## Checklist:

- [x] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

### CODE REVIEW

File: src/PrismicNextLink.tsx [hunk 1]
```diff
@@ -51,5 +51,14 @@ export const PrismicNextLink = React.forwardRef<
 		rel = restProps.rel;
 	}
 
-	return <Link ref={ref} {...attrs} {...restProps} href={href} rel={rel} />;
+	let text: string | undefined;
+	if (field && "text" in field) {
+		text = field.text;
+	}
```
Comment (angeloashmore) [inline src/PrismicNextLink.tsx:57]: See https://github.com/prismicio/prismic-react/pull/205#discussion_r1735434752

File: src/PrismicNextLink.tsx [hunk 2]
```diff
@@ -51,5 +51,14 @@ export const PrismicNextLink = React.forwardRef<
...
+
+	return (
+		<Link ref={ref} {...attrs} {...restProps} href={href} rel={rel}>
+			{children || text}
```
Comment (angeloashmore) [inline src/PrismicNextLink.tsx:61]: See https://github.com/prismicio/prismic-react/pull/205#discussion_r1735435335

File: test/PrismicNextLink.test.tsx [hunk 3]
```diff
@@ -237,3 +237,35 @@ it("forwards ref", (ctx) => {
 
 	expect(ref).toHaveBeenCalledWith({ tagName: "a" });
 });
+
+it("renders a next/link with text and without children", (ctx) => {
```
Comment (angeloashmore) [inline test/PrismicNextLink.test.tsx:241]:
```suggestion
it("renders a next/link with text", (ctx) => {
```

File: test/PrismicNextLink.test.tsx [hunk 4]
```diff
@@ -237,3 +237,35 @@ it("forwards ref", (ctx) => {
...
+	const field = ctx.mock.value.link({
+		type: "Web",
+		model: { type: "Link", config: { text: { type: "Text" } } },
```
Comment (angeloashmore) [inline test/PrismicNextLink.test.tsx:244]: See https://github.com/prismicio/prismic-react/pull/205#discussion_r1735437409

File: test/PrismicNextLink.test.tsx [hunk 5]
```diff
@@ -237,3 +237,35 @@ it("forwards ref", (ctx) => {
...
+	const expected = renderJSON(
+		<Link href={field.url} rel="noreferrer">
+			{field.text}
+		</Link>,
+	);
+
+	expectLinkToEqual(actual, expected);
+});
+
+it("renders a next/link with text and children", (ctx) => {
```
Comment (angeloashmore) [inline test/PrismicNextLink.test.tsx:257]:
```suggestion
it("renders a next/link with the given children, overriding the link's text", (ctx) => {
```

**Review by angeloashmore — CHANGES_REQUESTED** [review]
Looks good! I linked to a few suggestions from the `@prismicio/react` PR (https://github.com/prismicio/prismic-react/pull/205).

Like the `@prismicio/react` PR, I'm going to mark the PR as "Request changes" because we should publish `@prismicio/client` first and update the version here before merging. Let me know if you'd like to handle that a different way though.

**Comment by angeloashmore** [comment]
@xrutayisire The test was failing because the `xru/support-link-text` branch was not up to date with `main`. It was missing a fix @levimykel and I made here: https://github.com/prismicio/prismic-next/pull/94/commits/05510c921178d45b1489b310c592ba5fe62d3340

I knew I had seen the error before! 😄

File: src/PrismicNextLink.tsx (hunk 2 above)
Comment (xrutayisire) [reply src/PrismicNextLink.tsx:61]: Replied there 🙂

File: test/PrismicNextLink.test.tsx (hunk 5 above)
Comment (xrutayisire) [reply test/PrismicNextLink.test.tsx]: Done, thanks, better 🙂

File: test/PrismicNextLink.test.tsx (hunk 3 above)
Comment (xrutayisire) [reply test/PrismicNextLink.test.tsx]: Done, thanks, better 🙂

File: src/PrismicNextLink.tsx (hunk 1 above)
Comment (xrutayisire) [reply src/PrismicNextLink.tsx]: Thanks, done!

`const text = prismic.isFilled.link(field) ? field.text : undefined;`

File: src/PrismicNextLink.tsx (hunk 2 above)
Comment (angeloashmore) [reply src/PrismicNextLink.tsx:57]: Once we resolve the conversation there, I will approve this PR. 🙂

Comment (xrutayisire) [reply src/PrismicNextLink.tsx]: Done 👍

**Review by angeloashmore — APPROVED** [review]
Looks good!

File: test/createLocaleRedirect.test.ts [hunk 6]
```diff
@@ -49,7 +49,7 @@ it("creates a redirect with the request's preferred locale", async () => {
 
 	mockGetRepository({
 		client,
-		languages: [{ id: "fr-fr", name: "French (France)" }],
+		languages: [{ id: "fr-fr", name: "French (France)", is_master: true }],
```
Comment (angeloashmore) [inline test/createLocaleRedirect.test.ts:52]: 🎉 #nice: Thanks for updating these tests.

**Review by angeloashmore — APPROVED** [review]
Looks good!

**Pull request merged**